### PR TITLE
Code improvements

### LINKED ISSUE #116: As a UNDAC Software Engineer, the codebase needs to be in a state which allows development of features
End user goal: To have new features developed faster and with less defects

End business goal: To have new features developed faster and with less defects

Acceptance criteria:
[ ] Code duplication is lessened
[ ] Documentation and comment coverage is improved

Notes:
N/A

### PR DESCRIPTION
**Pull request by Pannoniae**
Closes #116 .

### CODE REVIEW

**Comment by ryanm2711** [comment]
Lgtm!

**Review by justinwylie033 — APPROVED** [review]
I have reviewed the recent changes made to the model properties and the refactoring of the SetProperty method to utilise a centralised Utils.SetProperty. Here are my observations and recommendations:

Observations:

Non-Nullable Fields Initialisation: Some model classes (Skill, SystemType, TeamMember) have non-nullable fields that are not guaranteed to be initialised within the constructor. This could potentially lead to null reference exceptions at runtime.
Event Hiding in Continent Model: The PropertyChanged event in the Continent model is hiding the inherited member from AModel. This may not be intentional and could cause event handling issues.
Nullable Reference Types Compliance: The project now has Nullable Reference Types enabled. Some methods, such as Utils.SetProperty, may not fully adhere to the new NRT rules, which could result in warnings or unexpected behaviour.

in conclusion, these changes are extremely beneficial and must be merged, however, more refactoring may be needed for universal best practice compliance
